src/io/flutter/FlutterUtils.java

```diff
@@ -440,4 +444,84 @@ protected void addImplicitResolvers() {
       return null;
     }
   }
+
+  public static boolean isAndroidxProject(@NotNull Project project) {
+    @SystemIndependent String basePath = project.getBasePath();
+    assert basePath != null;
+    VirtualFile projectDir = LocalFileSystem.getInstance().findFileByPath(basePath);
+    assert projectDir != null;
+    VirtualFile androidDir = getFlutterManagedAndroidDir(projectDir);
+    if (androidDir == null) {
+      return false;
+    }
+    VirtualFile properties = androidDir.findChild("gradle.properties");
+    if (properties == null) {
+      return false;
+    }
+    try {
+      List<String> lines = FileUtil.loadLines(properties.getPath(), "UTF-8");
+      for (String line : lines) {
+        if (line.trim().isEmpty() || line.startsWith("#")) continue;
+        int equalsIndex = line.indexOf('=');
+        if (equalsIndex > 0 && equalsIndex < line.length() - 1) {
+          String key = line.substring(0, equalsIndex).trim();
+          if ("android.useAndroidX".equals(key)) {
+            String value = line.substring(equalsIndex + 1).trim();
+            return Boolean.parseBoolean(value);
+          }
+        }
+      }
+    }
+    catch (IOException e) {
+      return false;
+    }
+    return false;
+  }
+
+  @Nullable
+  private static VirtualFile getFlutterManagedAndroidDir(VirtualFile dir) {
+    VirtualFile meta = dir.findChild(".metadata");
+    if (meta != null) {
+      try {
+        List<String> lines = FileUtil.loadLines(meta.getPath(), "UTF-8");
+        for (String line : lines) {
+          if (line.trim().isEmpty() || line.startsWith("#")) continue;
```

Comment:
same as above, switch to using org.gradle.wrapper.SystemPropertiesHandler if possible.

Reply:
Done.